[vulkan] Remove constant duplication for Vulkan optimize_for_mobile

IvanKobzarev opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#59276 [vulkan] Remove constant duplication for Vulkan optimize_for_mobile**

Differential Revision: [D28814072](https://our.internmc.facebook.com/intern/diff/D28814072)

IvanKobzarev committed:
[vulkan] Remove constant duplication for Vulkan optimize_for_mobile

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28814072).

cccclai commented:
LGTM

cccclai commented on torch/csrc/jit/passes/vulkan_rewrite.cpp:
```diff
@@ -213,6 +214,11 @@ void vulkanRemoveMutation(script::Module& module) {
 }
 
+void vulkanRunCanonicalOptimizations(script::Module& module) {
+  auto graph = module.get_method("forward").graph();
```
I think we supports other optimization methods for CPU. What about vulcan?

IvanKobzarev replied:
Yes, we should do the same for vulkan.

IvanKobzarev committed:
Update on "[vulkan] Remove constant duplication for Vulkan optimize_for_mobile"


Differential Revision: [D28814072](https://our.internmc.facebook.com/intern/diff/D28814072)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28814072).

cccclai commented on torch/csrc/jit/passes/vulkan_rewrite.cpp:
```diff
@@ -213,6 +214,14 @@ void vulkanRemoveMutation(script::Module& module) {
 }
 
+void vulkanRunCanonicalOptimizations(script::Module& module) {
+  auto graph = module.get_method("forward").graph();
```
This `graph` is not used.

IvanKobzarev committed:
Update on "[vulkan] Remove constant duplication for Vulkan optimize_for_mobile"


Differential Revision: [D28814072](https://our.internmc.facebook.com/intern/diff/D28814072)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28814072).
